### PR TITLE
Don't unwrap a tree that might not be there

### PR DESCRIPTION
**Pull request by AloeareV**

### CODE REVIEW

File: zingolib/src/wallet.rs [hunk 1]
```diff
@@ -251,17 +251,15 @@ impl LightWallet {
                 &[u8],
                 COMMITMENT_TREE_LEVELS,
             >(&hex::decode(trees.sapling_tree).unwrap()[..])
-            .unwrap()
-            .to_frontier()
-            .take(),
+            .ok()
```
[the remark this answers is not in the document]
Comment (zancas) [reply zingolib/src/wallet.rs:254]: My thought experiment is this:

If we had the **right** name in place, when this was first written would it have become obvious to the developer that a critical edge case was being overlooked?

Also....   I notice that we're replacing an `unwrap`..  this makes me skeptical about a previously held belief that "all the unwraps were reasonabile".

**Review by zancas — APPROVED** [review]
@fluidvanadium can you make the names **make** sense?   What's this code doing?  Like maybe the original `fn` should have been more carefully named:

`fn get_legacy_frontiers_except_in_real_world_edge_case_when_tree_is_empty`

**Review by zancas — CHANGES_REQUESTED** [review]
Let's ask @AArnott to test this.

**Comment by zancas** [comment]
I converted this to a draft when @AArnott 's (manual) test failed.

**Comment by zancas** [comment]
@fluidvanadium @Oscar-Pepper @AloeareV @AArnott do we have a test that shows that this change to our `dev` branch fixes something?

How hard would it be to produce such a test?

**Comment by AArnott** [comment]
Yes, it should be pretty straightforward for me to write the rust code that reproduces the failure I was seeing. I've never written a test in this repo before so the hardest part may be figuring out where and how to best do it.
Also, my test would initially express a dependency on a live lightwalletd server. If you avoid such network dependencies in preference to locally hosted servers, that's fine, but I don't know how to set that up and I'm not certain it'll repro in that configuration.

**Review by fluidvanadium — APPROVED** [review]
yup

**Comment by fluidvanadium** [comment]
fixes https://github.com/zingolabs/zingolib/issues/520
i believe that issue was introduced by a librustzcash update

**Comment by fluidvanadium** [comment]
> Yes, it should be pretty straightforward for me to write the rust code that reproduces the failure I was seeing. I've never written a test in this repo before so the hardest part may be figuring out where and how to best do it. Also, my test would initially express a dependency on a live lightwalletd server. If you avoid such network dependencies in preference to locally hosted servers, that's fine, but I don't know how to set that up and I'm not certain it'll repro in that configuration.

ive found it difficult to repro with locally hosted servers. connecting to the live server would be fine im sure. i would love to see this test

**Comment by zancas** [comment]
@fluidvanadium @Oscar-Pepper @AloeareV @AArnott do we have a test that shows that this change to our `dev` branch fixes something?

How hard would it be to produce such a test?

**Pull request merged**